### PR TITLE
Feature: dynamic schema actions

### PR DESCRIPTION
**Pull request by MathiasReichardt**
- New Action available `DynamicHypermediaAction` which is used for runtime parametrized actions
- Fix missing PUT action handling in ApplicationModel

### CODE REVIEW

File: Source/RESTyard.AspNetCore/WebApi/Formatter/SirenConverter.cs [hunk 1]
```diff
@@ -251,15 +253,41 @@ private void AddJsonParameterActionDescription(JObject jAction, HypermediaAction
             jAction.Add("fields", new JArray { jfield });
         }
 
+        private dynamic GetRoutKeysIfActionHasSchemaParameters(HypermediaActionBase hypermediaAction)
```
Comment (JMarianczuk) [inline Source/RESTyard.AspNetCore/WebApi/Formatter/SirenConverter.cs:256]: typo: Route

File: README.md [hunk 2]
```diff
@@ -504,6 +504,65 @@ The corresponding post would look like:
 ]
 ```
 
+### Dynamic Actions
+
+In certain scenarios an actions parameters (or not having any) is determined at runtime. This special case can be implemented using `DynamicHypermediaAction`.
+This action allows for a runtime dynamic schema to be retrieved. To decide what schema is desired it is required to pass runtime values to the schema routes.
+`DynamicHypermediaAction` has the property `SchemaRouteKeys` which accepts and object which will be passed to route generation so route keys can be filled with values.
+This also requires a custom route for the dynamic schemas using `[HttpGetHypermediaActionParameterInfo]`.
+For tis to work the custom type route has to:
```
Comment (JMarianczuk) [inline README.md:513]: typo: this

File: README.md [hunk 3]
```diff
@@ -504,6 +504,65 @@ The corresponding post would look like:
 ]
 ```
 
+### Dynamic Actions
+
+In certain scenarios an actions parameters (or not having any) is determined at runtime. This special case can be implemented using `DynamicHypermediaAction`.
+This action allows for a runtime dynamic schema to be retrieved. To decide what schema is desired it is required to pass runtime values to the schema routes.
+`DynamicHypermediaAction` has the property `SchemaRouteKeys` which accepts and object which will be passed to route generation so route keys can be filled with values.
```
Comment (JMarianczuk) [inline README.md:511]: typo: which accepts an object

**Review by JMarianczuk — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (MathiasReichardt) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (MathiasReichardt) [reply README.md]: Done

**Pull request merged**